#solucion parcial corte 2
## Juan Camilo Carranza Barrios
### Economia circular y valoracion ambiental
#### 2292847

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm


In [ ]:
df = pd.read_csv("recdemand.csv")

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 659 entries, 0 to 658
Data columns (total 8 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   trips    659 non-null    int64  
 1   quality  659 non-null    int64  
 2   ski      659 non-null    object 
 3   income   659 non-null    int64  
 4   userfee  659 non-null    object 
 5   costC    659 non-null    float64
 6   costS    659 non-null    float64
 7   costH    659 non-null    float64
dtypes: float64(3), int64(3), object(2)
memory usage: 41.3+ KB


In [ ]:
df=df[["trips","quality","ski","income","userfee","costC","costS","costH"]]

In [ ]:
df

trips  quality  ski  income userfee  costC   costS   costH
0        0        0  yes       4      no  67.59  68.620  76.800
1        0        0   no       9      no  68.86  70.936  84.780
2        0        0  yes       5      no  58.12  59.465  72.110
3        0        0   no       2      no  15.79  13.750  23.680
4        0        0  yes       3      no  24.02  34.033  34.547
..     ...      ...  ...     ...     ...    ...     ...     ...
654     40        5  yes       5     yes  25.63  18.916  29.684
655     40        3  yes       3      no  18.46  13.330  25.800
656     40        2  yes       4      no  58.84  47.000  62.760
657     50        4  yes       4      no  35.67  25.694  37.269
658     88        2   no       2      no  22.49   6.210  25.460

[659 rows x 8 columns]

In [ ]:
df = df.replace({"ski":{"yes": 1,"no":0},"userfee":{"yes":1,"no":0}})

In [ ]:
y = df["trips"]
x = df[["quality","ski","userfee","income","costC","costS","costH"]]
x = sm.add_constant(x)
model = sm.OLS(y, x)
reg = model.fit()
reg.summary()

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:117: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  trips   R-squared:                       0.279
Model:                            OLS   Adj. R-squared:                  0.271
Method:                 Least Squares   F-statistic:                     36.02
Date:                Fri, 22 Apr 2022   Prob (F-statistic):           1.29e-42
Time:                        14:37:41   Log-Likelihood:                -2038.8
No. Observations:                 659   AIC:                             4094.
Df Residuals:                     651   BIC:                             4130.
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.5615      0.572      4.479      0.000       1.439       3.685
quality        0.9134      0.125      7.305      0.000       0.668       1.159
ski            0.9992      0.462      2.161      0.031       0.091       1.907
userfee        8.2808      1.543      5.368      0.000       5.251      11.310
income        -0.2071      0.119     -1.734      0.083      -0.442       0.027
costC          0.0632      0.034      1.841      0.066      -0.004       0.131
costS         -0.1645      0.023     -7.236      0.000      -0.209      -0.120
costH          0.0895      0.028      3.177      0.002       0.034       0.145
==============================================================================
Omnibus:                      925.235   Durbin-Watson:                   0.378
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           238996.003
Skew:                           7.395   Prob(JB):                         0.00
Kurtosis:                      95.115   Cond. No.                         938.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

**1**. Modelo de regresión lineal que explique el número de viajes a partir del resto de las variables encontradas en la base de datos adjunta.
2. Conclusión sobre la significancia estadística de cada variable.
3. Interpretación de los coeficientes.

El diccionario de la base de datos es el siguiente:

trips : número de viajes en bote realizados en el lago Somerville

quality : percepción subjetiva de la calidad de 1 a 5

ski factor: pregunta acerca de si el individuo hizo o no water-ski en el lago Somerville

income: ingreso anual en miles de dólares 

userfee factor: pago de suscripción anual al lago 

costC: costo de visitar Lago Conroe (alternativa al lago Somerville) 

costS: costo de visitar Lago Somerville 

costH: costo de visitar Lago Houston (alternativa al lago Somerville) 

# Analisis

##Analisis significancia
Las variables significativas de este modelo son, quality, ski, userfee, costS, costH ya que el valor de su P>|t es inferior a 0.5**, caso contrario de las variables income y costC.

En el caso de las variables que presentaron nivel de significancia menor a 0.5,se rechaza la hipotesis nula, este resultado tambien representa el riesgo de tener diferencias entre el modelo y la hipotesis.

##Analisis coeficientes 
En el caso de los coeficientes dados por el modelo de regresion lineal, se puede evidenciar que solo hay dos valores que reducen la probabilidad de viajar, las cuales son el "income" (ingreso anual en miles de dolares)con un coeficiente de -0.2071 y la otra variable es "costoS" (costo de visitar lago somerville) con un coeficiente de -0.1645, esta variable de costosS al ser significante, se entiende que si influye en la probabilidad de viajar en la persona.las variables las cuales aumentan la probabilidad de viajar son quality, ski factor,userfee, costoC y costoH , sin embargo las variables "income y costoC"no tienen significancia es decir hay un riego alto de diferencia entre el modelo y la hipotesis.
